app/src/main/java/org/wikipedia/views/EditNoticesDialog.kt

```diff
@@ -0,0 +1,72 @@
+package org.wikipedia.views
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import org.wikipedia.R
+import org.wikipedia.databinding.DialogEditNoticesBinding
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.page.LinkMovementMethodExt
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.DimenUtil
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.UriUtil
+import org.wikipedia.util.log.L
+
+@SuppressLint("ViewConstructor")
+class EditNoticesDialog constructor(
+        val wikiSite: WikiSite,
+        val editNotices: List<String>,
+        context: Context
+) : AlertDialog(context) {
+
+    private val binding = DialogEditNoticesBinding.inflate(LayoutInflater.from(context))
+
+    private val movementMethod = LinkMovementMethodExt { urlStr ->
+        L.v("Link clicked was $urlStr")
+        UriUtil.visitInExternalBrowser(context, Uri.parse(UriUtil.resolveProtocolRelativeUrl(wikiSite, urlStr)))
```

Comment:
Would it be possible to implement the `LinkHandler` here so that some links can be opened in the app? e.g. talk pages or Wikipedia articles.

Reply:
This is actually done deliberately. If a link goes to a regular article, it will open `PageActivity`, and because PageActivity is `singleTask`, it will automatically close the Editing activity, and the user will lose their place, and won't be able to go back to editing.

Reply:
Ah! Right, that makes sense.